### PR TITLE
python3 3.6.1

### PR DESCRIPTION
**Pull request by neutric**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/python3.rb [hunk 1]
```diff
@@ -33,15 +32,20 @@ class Python3 < Formula
   skip_clean "bin/easy_install3", "bin/easy_install-3.4", "bin/easy_install-3.5", "bin/easy_install-3.6"
 
   resource "setuptools" do
-    url "https://files.pythonhosted.org/packages/87/1a/33d3d05569e857c5c5cc3e90d197bf4d9696dc740a05f66a09599d66e5bd/setuptools-32.2.0.zip"
-    sha256 "634313924fd186a2be0489c96965f5a909b666bd652eb3e16724913c707ec33f"
+    url "https://files.pythonhosted.org/packages/b5/e2/cca325e160b8e12334f5b2923ffcdb3c02da9fded1d2899d7930ac1c7448/setuptools-34.3.2.zip"
```
Comment (tdsmith) [inline Formula/python3.rb]: We'll see what CI says but I'm pretty sure we have to change our setuptools/pip bootstrapping strategy to use ensurepip if we upgrade beyond 32.2; I can work on this later if you want to leave setuptools alone for now

Comment (neutric) [inline Formula/python3.rb]: I undid my changes and will happily leave the setuptools update to you. :) For reference, this was the [outcome of the CI run](https://bot.brew.sh/job/Homebrew%20Core/19004/) based on these changes:

```diff
diff --git a/Formula/python3.rb b/Formula/python3.rb
index 30993e5cd..a22a32524 100644
--- a/Formula/python3.rb
+++ b/Formula/python3.rb
@@ -1,9 +1,8 @@
 class Python3 < Formula
   desc "Interpreted, interactive, object-oriented programming language"
   homepage "https://www.python.org/"
-  url "https://www.python.org/ftp/python/3.6.0/Python-3.6.0.tar.xz"
-  sha256 "b0c5f904f685e32d9232f7bdcbece9819a892929063b6e385414ad2dd6a23622"
-  revision 1
+  url "https://www.python.org/ftp/python/3.6.1/Python-3.6.1.tar.xz"
+  sha256 "a01810ddfcec216bcdb357a84bfaafdfaa0ca42bbdaa4cb7ff74f5a9961e4041"
   head "https://github.com/python/cpython", :using => :git

   bottle do
@@ -33,8 +32,8 @@ class Python3 < Formula
   skip_clean "bin/easy_install3", "bin/easy_install-3.4", "bin/easy_install-3.5", "bin/easy_install-3.6"

   resource "setuptools" do
-    url "https://files.pythonhosted.org/packages/87/1a/33d3d05569e857c5c5cc3e90d197bf4d9696dc740a05f66a09599d66e5bd/setuptools-32.2.0.zip"
-    sha256 "634313924fd186a2be0489c96965f5a909b666bd652eb3e16724913c707ec33f"
+    url "https://files.pythonhosted.org/packages/b5/e2/cca325e160b8e12334f5b2923ffcdb3c02da9fded1d2899d7930ac1c7448/setuptools-34.3.2.zip"
+    sha256 "4efcdcbc0b268a4c5b1094b36dd5ed55658968195577ed331cec9b1c71891f31"
   end

   resource "pip" do
@@ -42,6 +41,11 @@ class Python3 < Formula
     sha256 "09f243e1a7b461f654c26a725fa373211bb7ff17a9300058b205c61658ca940d"
   end

+  resource "six" do
+    url "https://files.pythonhosted.org/packages/b3/b2/238e2590826bfdd113244a40d9d3eb26918bd798fc187e2360a8367068db/six-1.10.0.tar.gz"
+    sha256 "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a"
+  end
+
   resource "wheel" do
     url "https://files.pythonhosted.org/packages/c9/1d/bd19e691fd4cfe908c76c429fe6e4436c9e83583c4414b54f6c85471954a/wheel-0.29.0.tar.gz"
     sha256 "1ebb8ad7e26b448e9caa4773d2357849bf80ff9e313964bcaf79cbf0201a1648"
@@ -186,7 +190,7 @@ class Python3 < Formula
     # Remove the site-packages that Python created in its Cellar.
     (prefix/"Frameworks/Python.framework/Versions/#{xy}/lib/python#{xy}/site-packages").rmtree

-    %w[setuptools pip wheel].each do |r|
+    %w[six setuptools pip wheel].each do |r|
       (libexec/r).install resource(r)
     end

@@ -227,7 +231,7 @@ class Python3 < Formula
     rm_rf Dir["#{site_packages}/distribute*"]
     rm_rf Dir["#{site_packages}/pip[-_.][0-9]*", "#{site_packages}/pip"]

-    %w[setuptools pip wheel].each do |pkg|
+    %w[six setuptools pip wheel].each do |pkg|
       (libexec/pkg).cd do
         system bin/"python3", "-s", "setup.py", "--no-user-cfg", "install",
                "--force", "--verbose", "--install-scripts=#{bin}",
@@ -318,8 +322,8 @@ class Python3 < Formula
       xy = version.to_s.slice(/(3\.\d)/) || "3.6"
     end
     text = <<-EOS.undent
-      Pip, setuptools, and wheel have been installed. To update them
-        pip3 install --upgrade pip setuptools wheel
+      Pip, setuptools, six, and wheel have been installed. To update them
+        pip3 install --upgrade pip setuptools six wheel

       You can install Python packages with
         pip3 install <package>
```

Comment (tdsmith) [inline Formula/python3.rb]: Thank you!

**Comment by neutric** [comment]
@tdsmith All yours!

**Pull request closed without merging**